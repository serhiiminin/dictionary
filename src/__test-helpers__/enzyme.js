import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
}

const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  })
}

export { shallowWithTheme, mountWithTheme }
