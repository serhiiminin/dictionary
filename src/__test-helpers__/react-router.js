const mockRouterProps = {
  match: {
    isExact: true,
    params: {},
    path: "",
    url: ""
  },
  location: {
    hash: "",
    key: "",
    pathname: "",
    search: "",
    state: {}
  },
  history: {
    length: 2,
    action: "POP",
    location: location,
    push: () => {},
    replace: () => {},
    go: () => {},
    goBack: () => {},
    goForward: () => {},
    block: () => {},
    createHref: () => {},
    listen: () => {}
  },
  staticContext: {}
};

export default mockRouterProps;
