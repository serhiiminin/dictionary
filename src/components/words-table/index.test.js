import React from "react";
import { MemoryRouter as Router } from "react-router-dom";
import muiTheme from "../../root/mui-theme";
import { mountWithTheme, shallowWithTheme } from "../../__test-helpers__/enzyme";
import mockRouterProps from "../../__test-helpers__/react-router";
import WordTable, { WordsListWrapper } from "./component";

describe("WordTable components", () => {
  test("WordsListWrapper", () => {
    const wrapper = shallowWithTheme(
      <WordsListWrapper>Anything</WordsListWrapper>,
      muiTheme
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe("WordForm", () => {
  let wrapper;
  const wordsList = [
    { _id: '1' },
    { _id: '2' },
  ];
  beforeEach(() => {
    wrapper = mountWithTheme(
      <Router
        initialEntries={["/one", "/two", { pathname: "/three" }]}
        initialIndex={1}
      >
        <WordTable
          wordsList={wordsList}
          wordsCount={2}
          history={mockRouterProps.history}
          location={mockRouterProps.location}
          getWordsSearchParams={() => ({ countPerPage: 5, sortBy: 'dateCreated', sortDirection: 'ascend', page: 1})}
          deleteWord={() => {}}
          checkIsLoading={() => {}}
        >
          Anything
        </WordTable>
      </Router>,
      muiTheme
    );
  });
  test("render", () => {
    expect(wrapper).toEqual({});
  });
});
