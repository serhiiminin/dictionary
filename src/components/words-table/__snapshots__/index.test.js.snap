// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordTable components WordsListWrapper 1`] = `
<div
  className="sc-eHgmQL bYYnQm"
>
  Anything
</div>
`;
